feat: APIM - Added additional types and UDTs

AlexanderSehr opened the pull request:
## Description

As per issue added
- `Microsoft.ApiManagement/service/apis/operations` 
- `Microsoft.ApiManagement/service/apis/operations/policies`

Added UDTs & RDTs for several resources
Aligned interfaces to specs

## Pipeline Reference

<!-- Insert your Pipeline Status Badge below -->

| Pipeline |
| -------- |
|          |

## Type of Change

<!-- Use the checkboxes [x] on the options that are relevant. -->

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

eriqua commented on utilities/pipelines/e2eValidation/resourceDeployment/New-TemplateDeployment.ps1:
```diff
@@ -428,7 +428,7 @@ function New-TemplateDeployment {
 
         [Parameter(Mandatory = $false)]
-        [int] $RetryLimit = 3,
+        [int] $RetryLimit = 1, # TODO: Set back to 3
```
Todo to do

AlexanderSehr replied:
Addressed. Thanks :)

eriqua commented:
@AlexanderSehr 1 small one for the CI for now, let me know if you need a review for the rest

AlexanderSehr commented:
> @AlexanderSehr 1 small one for the CI for now, let me know if you need a review for the rest

Thank you. Addressed your comment. I think for the rest it would be best to have the module owner @tony-box review the changes (unless he does not have the capacity. @tony-box, let us know if the case)

tony-box commented on avm/res/api-management/service/api/operation/main.bicep:
```diff
@@ -0,0 +1,97 @@
+metadata name = 'API Management Service APIs Policies'
```
Is this metadata description correct?

AlexanderSehr replied:
Good catch @tony-box. Fixing...

AlexanderSehr replied:
Done

tony-box reviewed (changes requested):
Looks good other than my one minor comment about the description. Thanks @AlexanderSehr!

eriqua commented:
> > @AlexanderSehr 1 small one for the CI for now, let me know if you need a review for the rest
> 
> Thank you. Addressed your comment. I think for the rest it would be best to have the module owner @tony-box review the changes (unless he does not have the capacity. @tony-box, let us know if the case)

Sounds good. @tony-box, once you are OK with the review just ping me so that I can unblock the approval

eriqua commented on utilities/pipelines/sharedScripts/helper/Get-SpecsAlignedResourceName.ps1:
```diff
@@ -104,8 +104,12 @@ function Get-SpecsAlignedResourceName {
                 # Setting explicitly as both [apimanagement/service/apis/policies] & [apimanagement/service/apis/policy] exist in the specs and the later seem to have been an initial incorrect publish (only one API version exists)
                 $resourceType = 'service/apis/policies'
             }
+            'service/api/operation/policy' {
```
disable actions
